util.go

```diff
@@ -44,18 +44,12 @@ func requestJob(id uuid.UUID, ch chan jobOutRequest) *internalJob {
 
 func requestJobCtx(ctx context.Context, id uuid.UUID, ch chan jobOutRequest) *internalJob {
```

Comment:
IMO, this function should throw an error if the context deadline is exceeded instead of returning `nil`.

Reply:
yeah that seems reasonable. A context deadline exceeded error?